From fcc9a80e47f1a9945987e398626074bddfc705c0 Mon Sep 17 00:00:00 2001
From: Yongxin Liu <yongxin.liu@windriver.com>
Date: Fri, 24 Jul 2020 09:27:48 +0800
Subject: [PATCH] qat: replace linux/cryptohash.h with crypto/sha.h for kernel
 v5.8

This is to accommodate kernel commit 228c4f265c6e ("crypto:
lib/sha1 - fold linux/cryptohash.h into crypto/sha.h").

Upstream-Status: Inappropriate [Code released in tarball form only]

Signed-off-by: Yongxin Liu <yongxin.liu@windriver.com>
---
 .../utilities/osal/src/linux/kernel_space/OsalCryptoInterface.c       | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/quickassist/utilities/osal/src/linux/kernel_space/OsalCryptoInterface.c b/quickassist/utilities/osal/src/linux/kernel_space/OsalCryptoInterface.c
index 2d5b233..92ee35b 100644
--- a/quickassist/utilities/osal/src/linux/kernel_space/OsalCryptoInterface.c
+++ b/quickassist/utilities/osal/src/linux/kernel_space/OsalCryptoInterface.c
@@ -66,7 +66,11 @@
 
 #include "Osal.h"
 #include <linux/crypto.h>
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5,8,0))
+#include <crypto/sha.h>
+#else
 #include <linux/cryptohash.h>
+#endif
 #include <linux/version.h>
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,29))
 #include <crypto/internal/hash.h>
-- 
2.14.4

